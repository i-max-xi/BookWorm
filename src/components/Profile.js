import '../styles/profile.css';
import { useSelector } from 'react-redux';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets).filter(
    (rocket) => rocket.reserved,
  );
  const missions = useSelector((state) => state.missions).filter(
    (mission) => mission.reserved,
  );

  return (
    <div className="profilePage">
      <div className="filteredContainer">
        <h2>My Missions</h2>
        <ul className="filteredList">
          {missions.length > 0 ? (
            missions.map((mission) => (
              <li key={mission.name}>{mission.name}</li>
            ))
          ) : (
            <p key="message">You haven&apos;t joined any missions yet.</p>
          )}
        </ul>
      </div>
      <div className="filteredContainer">
        <h2>My Rockets</h2>
        <ul className="filteredList">
          {rockets.length > 0 ? (
            rockets.map((rocket) => <li key={rocket.name}>{rocket.name}</li>)
          ) : (
            <p key="message">You don&apos;t have any rockets yet.</p>
          )}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
