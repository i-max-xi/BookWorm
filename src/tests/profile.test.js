import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../Redux/store';
import Profile from '../components/Profile';

describe('test if profile page works correctly', () => {
  test('renders correctly', () => {
    const profile = renderer.create(
      <Provider store={configureStore}>
        <Profile />
      </Provider>,
    );
    expect(profile).toMatchSnapshot();
  });
});
