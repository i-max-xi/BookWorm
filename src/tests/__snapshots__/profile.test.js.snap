// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if profile page works correctly renders correctly 1`] = `
<div
  className="profilePage"
>
  <div
    className="filteredContainer"
  >
    <h2>
      My Missions
    </h2>
    <ul
      className="filteredList"
    >
      <p>
        You haven't joined any missions yet.
      </p>
    </ul>
  </div>
  <div
    className="filteredContainer"
  >
    <h2>
      My Rockets
    </h2>
    <ul
      className="filteredList"
    >
      <p>
        You don't have any rockets yet.
      </p>
    </ul>
  </div>
</div>
`;
